### DISCUSSION TITLE
Version 2.0.0 is released

### DISCUSSION THREAD

**Discussion by avm87** [krlove/eloquent-model-generator#89] [Announcements]
Dear all,

Version 2.0.0 is released. It contains several improvements, new features and breaking changes, which I'd like to guide you through.

Minimum supported versions - Laravel 9 and PHP 8.0.

One of the most requested features was generating models from all tables. Now it is possible with `krlove:generate:models` command. This command lists all tables, transforms their names into class names (e.g. `users` -> `User`) and generates Eloquent models for them. See more details in the [README](https://github.com/krlove/eloquent-model-generator/blob/2.0.0/README.md#generating-models-for-all-tables).

`backup` option is removed, `no-backup` which defaults to `false` has been added instead. That means backup is enabled by default for both commands. If you want to disable it, pass `--no-backup` as a command option, or override it globally in your `config/eloquent_model_generator.php` file:
```php
return [
    'no_backup' => true,
];
```

The structure on config file has been flattened.
Before:
```php
return [
    'model_defaults' => [
        'namespace' => 'App\MyModels',
        'base_class_name' => 'App\MyModels\MyBaseClass',
        'output_path' => '/var/www/html/app/my_models',
        'no_timestamps' => true,
        'date_format' => 'U',
        'connection' => 'secondary',
        'backup' => true,
    ],
    'db_types' => [
        'enum' => 'string',
    ],
];
```
After:
```php
return [
    'namespace' => 'App\MyModels',
    'base_class_name' => 'App\MyModels\MyBaseClass',
    'output_path' => '/var/www/html/app/my_models',
    'no_timestamps' => true,
    'date_format' => 'U',
    'connection' => 'secondary',
    'no_backup' => false,
    'db_types' => [
        'box' => 'string',
    ],
];
```

`enum` type cast has been included out of the box. It casts to `string`. You can always override it in your config file:
```php
return [
    'db_types' => [
        'enum' => 'integer',
    ],
];
```

Generic code improvements and simplifications have been done:
- added typings to all places where it was possible
- used constructor property promotion
- got rid/simplified few key classes
- change visibility of certain methods/properties, normally from `protected` to `private`

Last but not least, (finally!) added tests covering key functionality.

If you have troubles upgrading from version 1x to 2x, feel free to create an issue or drop a comment under this discussion.

Best of luck!

**Comment by exsterx** [comment]
Hi guy!
I try to do install your packet and get following error:

**Problem 1
    - Root composer.json requires krlove/eloquent-model-generator ^2.0 -> satisfiable by krlove/eloquent-model-generator[2.0].
    - krlove/eloquent-model-generator 2.0 requires illuminate/database ^9.0 -> found illuminate/database[v9.0.0-beta.1, ..., 9.x-dev] but these were not loaded, likely because it conflicts with another require.
**

thanks for your attention

**Reply by navidman** [reply]
same problem here

**Reply by bogdancrb** [reply]
That's because the package is not compatible with Laravel v10... as this version requires v9.0 of "illuminate/database", and the latest Laravel framework requires v10 of "illuminate/database". So only solution unfortunately for now is to downgrade Laravel to v9 in order to use this package, if necessary.

**Comment by carcinocron** [comment]
Does this fix the issue where:
```sh
php artisan krlove:generate:model Models/First/Second/Third --table-name=first_second_third
```
Generates:
```php
<?php

namespace App;

class Models/First/Second/Third extends Model {
```
instead of
```php
<?php

namespace App\Models\First\Second;

class Third extends Model {
```
and/or also fails if the folder `app\Models\First\Second` does not already exist?
